src/hal/delay.h

```diff
@@ -1,6 +1,8 @@
 #ifndef __DELAY_H
 #define __DELAY_H
 
+#include <stm32f10x.h>
```

Comment:
I needed this to get the project to compile. I am not really sure why (or more specifically, I don't understand why it works without this!)

Reply:
I don't see how the STM32 header would be needed by this header file? Maybe it's missing somewhere else!

Though this header does use the stdint types, so `stdint.h` should be there!

So I'd change this.